### PR TITLE
#609 #687 #688 - env vars in configmap and x-robots-tag on autogenerated routes

### PR DESCRIPTION
**Pull request by twardnw**
This solves tickets #609 #687 #688
#687 is adding the openshift name to the configmap
#688 is adding the autogenerated route names in a configmap
#609 sets the x-robots-tag header on autogenerated routes to prevent crawler indexing

### CODE REVIEW

File: images/nginx/helpers/x-robots-header-development.conf [hunk 1]
```diff
@@ -1,13 +1,9 @@
-# Load environment variable LAGOON_ENVIRONMENT_TYPE into nginx variable. LAGOON_ENVIRONMENT_TYPE will be replaced via envplate during docker-entrypoint
-set $lagoon_environment_type '${LAGOON_ENVIRONMENT_TYPE:-development}';
-
-# By default no x-robots-tag
-set $x_robots_tag '';
-
-# If this is an development environment do not index
-if ($lagoon_environment_type = 'development') {
-  set $x_robots_tag 'noindex, nofollow';
+# Load environment variable LAGOON_ENVIRONMENT_TYPE into nginx variable. LAGOON_ENVIRONMENT_TYPE will be replaced via envplate during docker-entrypo
```
Comment (Schnitzel) [inline images/nginx/helpers/x-robots-header-development.conf]: as LUA actually has access to the env variables itself, we could now load the env variables directly instead via the envplate system in the docker-entrypoint which is hard to understand.

be careful though with: https://stackoverflow.com/questions/41800071/unable-to-use-environment-variables-in-lua-code

File: images/nginx/helpers/x-robots-header-development.conf [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
+header_filter_by_lua_block {
+  host = string.gsub(ngx.var.host, "%p", "%%%1")
```
Comment (Schnitzel) [inline images/nginx/helpers/x-robots-header-development.conf:4]: can you document what this does?

**Review by Schnitzel — CHANGES_REQUESTED** [review]
overall super awesome work, thanks! Just some improvements :)

File: images/nginx/helpers/x-robots-header-development.conf (hunk 2 above)
Comment (twardnw) [reply images/nginx/helpers/x-robots-header-development.conf:4]: 👍🏻

**Review by Schnitzel — DISMISSED** [review]

File: images/nginx/helpers/x-robots-header-development.conf [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-
-# By default no x-robots-tag
-set $x_robots_tag '';
-
-# If this is an development environment do not index
-if ($lagoon_environment_type = 'development') {
-  set $x_robots_tag 'noindex, nofollow';
+# Set X-Robots-Tag to 'noindex, nofollow' for development environments and Lagoon autogenerated routes
+header_filter_by_lua_block {
+  -- escape characters in the hostname
```
Comment (Schnitzel) [inline images/nginx/helpers/x-robots-header-development.conf:3]: can you describe which characters we are escaping?

**Review by Schnitzel — CHANGES_REQUESTED** [review]

**Pull request merged**